### ISSUE TITLE
[Feature] Autosave feed items to Internet Archive Wayback Machine

### ISSUE THREAD

**Issue by squromiv** [FreshRSS/FreshRSS#4929]
There is a problem of censorship and content disappearance from sites which I subscribed to in FreshRSS. We have [Wayback Machine Web Browser Extension](https://github.com/internetarchive/wayback-machine-webextension), that supports an option to save pages to Internet Archive, if they were not been archived previously. But in order to do it, we must open pages manually.

![screenshot 2022-12-08 12-13-14 001](https://user-images.githubusercontent.com/15985877/206406690-090d61fc-31d2-418d-a2ed-6bc18e8ec6c9.png)

It would be great to have such a feature for certain feeds in FreshRSS. Updating a feed also results to saving article pages to Internet Archive, if there isn`t any previously saved version. What do you think about it?

**Comment by daffydock** [comment]
Just to give my ¢.02, I think this would be a great option. I too come across site pages that I have visited that are either no longer there or the text has been changes after posting. Including known news sites. After the fact. 

I use an extension for it on Librefox but it would be awesome if this was included on FreshRSS. Since these days, FreshRSS is my main lead for news stories and the like.

**Comment by squromiv** [comment]
I`ve found a [service](https://github.com/plibither8/wayback) with similar function. But we need to send feed news urls to it somehow during updating.

**Comment by math-GH** [comment]
In 1.20.x the `archive.ph` is implemented as sharing service.

So you can save/archive the website you want.

Does it solve your feature request too?

**Comment by squromiv** [comment]
> Does it solve your feature request too?

Nope. Share feature needs manual mode, but we need to preserve items of "delicate" feeds automatically.

**Comment by math-GH** [comment]
I would prefer to have it as extension.

I do not see it relevant for the most of users to archive all articles. I have some fears that the FreshRSS instances will bomb the Archive project with thousands of archiving requests.

**Comment by squromiv** [comment]
> I would prefer to have it as extension.

May be. As well as an option for every feed to switch on/off such a function.

> I have some fears that the FreshRSS instances will bomb the Archive project with thousands of archiving requests.

Thousands of archiving requests are being sent from different users now.  Don`t know how archive.org deals with requests from the same IP, but it has API for similar purposes.

**Comment by Alkarex** [comment]
Related https://github.com/FreshRSS/FreshRSS/pull/5096

**Comment by yzqzss** [comment]
If you have a public instance of FreshRSS, try this python script I wrote earlier.
I'm just using it for my own use, so the code is ugly.
I've archived 1.5k+ RSS feeds with it.

https://gist.github.com/yzqzss/f9fc4dc3b58a601a0f9bd22fb47559f6

It might be more efficient to use the API or make a FreshRSS extension, but I'm lazy.

---

You can config SPN2 API parameters at L118:

![config](https://user-images.githubusercontent.com/30341059/224025847-b135df4d-a68d-440d-9faf-9c3fbb11d92f.png)

Save Page Now 2 Public API Docs: <https://docs.google.com/document/d/1Nsv52MvSjbLb2PCpHlat0gkzw0EvtSgpKHu4mk0MnrA>

---

screenshots:

![img1](https://user-images.githubusercontent.com/30341059/224023116-33af9719-7bba-424f-9404-b3ec1fc3c2c6.png)
![img2](https://user-images.githubusercontent.com/30341059/224023223-f9716374-3b9c-45ea-9aed-73f89dd79550.png)

**Comment by yzqzss** [comment]
You can use https://github.com/ArchiveBox/ArchiveBox to auto submit your feed articles into Internet Archive by using the FreshRSS RSS output (https://example.org/i/?a=rss&user=user&token=xxx&hours=168).

https://archivebox.io/#input-formats

https://github.com/ArchiveBox/ArchiveBox/wiki/Scheduled-Archiving#example-archive-all-urls-in-an-rss-feed-from-pocket-every-12-hours

**Comment by yzqzss** [comment]
You can use https://github.com/ArchiveBox/ArchiveBox to auto submit your feed articles into Internet Archive by using the FreshRSS RSS output (https://example.org/i/?a=rss&user=user&token=xxx&hours=168).

https://archivebox.io/#input-formats

https://github.com/ArchiveBox/ArchiveBox/wiki/Scheduled-Archiving#example-archive-all-urls-in-an-rss-feed-from-pocket-every-12-hours

**Comment by yzqzss** [comment]
> I do not see it relevant for the most of users to archive all articles. I have some fears that the FreshRSS instances will bomb the Archive project with thousands of archiving requests.

Their API has a rate limit: one IP can only save 6~7k pages per day. That's enough for us to take snapshot daily.

**Comment by squromiv** [comment]
@yzqzss
Thanks. Will check this info.

**Comment by Patriot2407** [comment]
I would also be interested in an extension for this, and what's more, being able to use archive.org api to check if article was already archived would be nice.
